packages/calypso-e2e/src/lib/components/jetpack-instant-search-modal-component.ts

```diff
@@ -0,0 +1,119 @@
+import { Locator, Page } from 'playwright';
+
+/**
+ * Component representing the modal popup for Jetpack Instant Search.
+ */
+export class JetpackInstantSearchModalComponent {
+	private page: Page;
+
+	/**
+	 * Constructs an instance of the component.
+	 *
+	 * @param {Page} page The underlying page.
+	 */
+	constructor( page: Page ) {
+		this.page = page;
+	}
+
+	/**
+	 * A locator to the parent modal dialog.
+	 */
+	get modalLocator(): Locator {
+		return this.page.getByRole( 'dialog', { name: 'Search results' } );
+	}
```

Comment:
**comment** - interesting, is this like the `anchor` or `block` that I have been using recently, but as a getter? Do you recommend doing this over the class attribute method I've been using?

Reply:
Yep! Since we're in a class, I thought I'd make use of class structure and just put the parent locator as an instance getter. It defers the execution until it's invoked, but also has that nice `this.modalLocator.getByX` syntax!

Reply:
> Do you recommend doing this over the class attribute method I've been using?

Oh, and to be clear, not necessarily! It's just different style. I think both are fine. 😄